modules/standard/Random.chpl

```diff
@@ -69,115 +68,152 @@ module Random {
   private use IO;
   private use Math;
 
+  @chpldoc.nodoc
+  record _RNG {
+    proc type PCG param do return 1;
+    proc type NPB param do return 2;
+  }
 
   /* Select between different supported RNG algorithms.
      See :mod:`PCGRandom` and :mod:`NPBRandom` for details on
      these algorithms.
    */
-  enum RNG {
-    PCG = 1,
-    NPB = 2
-  }
+  @deprecated("'RNG' is deprecated; please use the PCG algorithm via the :record:`randomStream` type or the NPB algorithm via the :mod:`NPBRand` package module")
```

Comment:
I would think that deprecating the enum would do it? Did you do the more complicated thing to work around something?

Reply:
I wanted to be able to use `_RNG` internally without getting deprecation warnings but the following caused the compiler to complain:
```chapel
@chpldoc.nodoc
enum _RNG {
...
}

@deprecated("...")
type RNG = _RNG;
```